packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -245,6 +269,33 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
      * @param _ethToDeposit total ETH that will be deposited in to the strategy which is a combination of msg.value and flash swap proceeds
      */
     function flashDeposit(uint256 _ethToDeposit) external payable nonReentrant {
+        uint256 wSqueethToMint = _flashDeposit(_ethToDeposit, msg.value);
+
+        emit FlashDeposit(msg.sender, _ethToDeposit, wSqueethToMint);
+    }
+
+    function flashDepositERC20(
+        uint256 _ethToDeposit,
+        uint256 _amountIn,
+        uint256 _minEthToGet,
+        uint24 _fee,
+        address _tokenIn
+    ) external nonReentrant {
+        uint256 wethReceived = _swapExactInputSingle(
+            _tokenIn,
+            weth,
+            msg.sender,
+            address(this),
+            _amountIn,
+            _minEthToGet,
+            _fee
+        );
+        uint256 wSqueethToMint = _flashDeposit(_ethToDeposit, wethReceived);
+
+        emit FlashDepositERC20(msg.sender, _tokenIn, _amountIn, _ethToDeposit, wSqueethToMint);
+    }
+
+    function _flashDeposit(uint256 _ethToDeposit, uint256 _ethInContract) internal returns (uint256) {
```

Comment:
why do you differentiate between `ethToDeposit` and `_ethInContract`? Not sure I follow the need for the separation. 

Why not just have `ethToDeposit`?

Reply:
@aparnakr  So `ethToDeposit` is the amount of ETH you want to deposit in total whereas `_ethInContract` is the amount you are actually spending. SO  `ethToDeposit - _ethInContract` Is the amount of ETH to flashloan.

We did not need _ethInContract before because you will just have ETH the user sent. But now in `flashDepositERC20 `, you will have `WETH` instead of ETH. Hence the 2nd param. 

As an option, I can convert them to ETH in `flashDepositERC20`. But anyway we are going to do that in `_strategyFlash` to convert WETH we get from flashloan. So don't want to do this 2 times